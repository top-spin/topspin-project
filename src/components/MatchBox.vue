<template>
  <div>
    <h4>Match</h4>
    <div v-if="match">
      <v-list-tile class="box" router :to="'/profile/'+match.match_winner.username">
        <v-list-tile-avatar class="hidden-md-only">
          <img :src="match.match_winner.avatar">
        </v-list-tile-avatar>

        <v-list-tile-title>{{match.match_winner.username}}</v-list-tile-title>
        <v-spacer></v-spacer>

        <v-list-tile-content class="score1">{{match.winning_score}}</v-list-tile-content>
      </v-list-tile>
      <v-list-tile class="box" router :to="'/profile/'+match.match_loser.username">
        <v-list-tile-avatar class="hidden-md-only">
          <img :src="match.match_loser.avatar">
        </v-list-tile-avatar>

        <v-list-tile-title>{{match.match_loser.username}}</v-list-tile-title>
        <v-spacer></v-spacer>

        <v-list-tile-content class="score1">{{match.losing_score}}</v-list-tile-content>
      </v-list-tile>
    </div>
  </div>
</template>

<script>
export default {
  name: "MatchBox",
  props: {
    match: Object
  },
  data() {
    return {
      players: [
        {
          image: "https://randomuser.me/api/portraits/men/85.jpg",
          username: "James",
          score: 19,
          route: "/profile"
        },
        {
          image: "https://randomuser.me/api/portraits/men/85.jpg",
          username: "Tyler",
          score: 12,
          route: "/profile"
        }
      ]
    };
  }
};
</script>

<style>
.box {
  border: #424242 3px solid;
}
.score1 {
  width: 25%;
  border-left: #999 2px solid;
  padding-left: 5px;
}
.score2 {
  border-left: #f57c00 3px solid;
  padding-left: 5px;
}
</style>
