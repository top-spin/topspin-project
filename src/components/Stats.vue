<template>
  <div>
    <span class="title">
      <span class="font-weight-light">My</span>
      <span class="orange--text">Statistics</span>
    </span>
    <span class="hidden-sm-and-down">
      <chartjs-doughnut
        v-if="datasets[0].data.length!==0"
        class="donutchart"
        :labels="labels"
        :datasets="datasets"
        :option="option"
      ></chartjs-doughnut>
    </span>

    <!-- horizontal-bars for smaller screen size -->
    <!-- TODO: Note. NOGO: datalabel -->
    <span class="hidden-md-and-up">
      <chartjs-horizontal-bar
        class="barchart"
        :labels="labels"
        :datasets="datasets"
        :datalabel="datalabel"
        :option="option"
      ></chartjs-horizontal-bar>
    </span>
  </div>
</template>

<script>
import axios from "axios";

export default {
  data() {
    return {
      labels: ["Tournament Wins", "Wins", "Losses"],
      datalabel: "Games Played",
      datasets: [
        {
          backgroundColor: ["#D32F2F", "#F57C00", "#FFD54F"],
          data: []
        }
      ],
      option: {
        title: {
          display: false,
          position: "bottom",
          text: "Wins to Losses"
        }
      }
    };
  },
  mounted() {
    axios
      .get("/api/get-my-stats")
      .then(res => {
        console.log("before==> ", this.datasets);
        this.datasets[0].data = res.data;
        console.log("after==> ", this.datasets);
      })
      .catch(err => console.log(err));
  }
};
</script>
 

<style>
.title {
  margin-top: 5px;
  display: flex;
  justify-content: center;
  /* border: blue 3px solid; */
}
.donutchart {
  height: 90%;
  /* border: green 3px solid; */
}
.barchart {
  height: 80%;
  /* border: red 3px solid; */
}
</style>