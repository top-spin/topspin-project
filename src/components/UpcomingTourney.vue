<template>
  <div>
    <span class="title">
      <span class="font-weight-light">Upcoming</span>
      <span class="orange--text">Tournaments</span>
    </span>
    <p>list all participating tourneys</p>
  </div>
</template>
