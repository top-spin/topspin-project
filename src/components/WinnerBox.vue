<template>
  <div v-if="match">
    <h4>Winner</h4>
    <v-list-tile
      class="box"
      router
      :to="'/profile/'+match.match_winner.username"
    >
      <v-list-tile-avatar>
        <img :src="match.match_winner.avatar">
      </v-list-tile-avatar>

      <v-list-tile-content>
        <v-list-tile-title>{{match.match_winner.username}}</v-list-tile-title>
      </v-list-tile-content>
    </v-list-tile>
  </div>
</template>

<script>
export default {
  name: "WinnerBox",
  props: {
    match:Object
  },
  data() {
    return {
      players: [
        {
          image: "https://randomuser.me/api/portraits/men/85.jpg",
          username: "James",
          score: 19,
          route: "/profile"
        }
      ]
    };
  }
};
</script>

<style>
.box {
  border: #424242 3px solid;
}
</style>
