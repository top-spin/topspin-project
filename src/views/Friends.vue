<template>
  <div>
                <v-radio-group light v-model="showfollowing" :mandatory="false">
              <v-radio color="primary" label="Following" :value="true"></v-radio>
              <v-radio color="primary" label="Followers" :value="false"></v-radio>
            </v-radio-group>
        <h1 v-if="showfollowing"  class="text-xs-left">Following</h1>
    <Followinglist v-if="showfollowing"/>

    
        <h1 v-if="!showfollowing" class="text-xs-left">Followers</h1>

        <Followerslist v-if="!showfollowing"  />

  </div>
</template>


<script>
import Followinglist from "@/components/Followinglist";
import Followerslist from "@/components/Followerslist";
export default {
  name: "Friends",
  components: {
    Followinglist,
    Followerslist
  },
  data() {
    return {
      showfollowing:true
    };
  }
};
</script>

<style>
</style>

