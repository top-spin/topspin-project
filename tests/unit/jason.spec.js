import { shallowMount } from "@vue/test-utils";
 
// sanity test
test("sanity test", () => {
  return;
});

//snapshot testing
test("renders correctly", () => {
  const wrapper = mount(Stats);
  expect(wrapper.element).toMatchSnapshot();
});

import { mount } from "@vue/test-utils";
import Friends from "@/views/Friends.vue";
  
import MatchBox from "@/components/MatchBox.vue";
import WinnerBox from "@/components/WinnerBox.vue";
import Stats from "@/components/Stats.vue";
import UpcomingTourney from "@/components/UpcomingTourney.vue";
 

describe("Friends", () => {
  it("is a Vue Instance", () => {
    const wrapper = mount(Friends);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
});
describe("MatchBox", () => {
  it("is a Vue Instance", () => {
    const wrapper = mount(MatchBox);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
});
describe("WinnerBox", () => {
  it("is a Vue Instance", () => {
    const wrapper = mount(WinnerBox);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
});
describe("Stats", () => {
  it("is a Vue Instance", () => {
    const wrapper = shallowMount(Stats);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
});
describe("UpcomingTourney", () => {
  it("is a Vue Instance", () => {
    const wrapper = shallowMount(UpcomingTourney);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
});
 