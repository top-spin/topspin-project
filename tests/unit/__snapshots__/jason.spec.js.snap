// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <span
    class="title"
  >
    <span
      class="font-weight-light"
    >
      My
    </span>
     
    <span
      class="orange--text"
    >
      Statistics
    </span>
  </span>
   
  <span
    class="hidden-sm-and-down"
  >
    <!---->
  </span>
   
  <!---->
</div>
`;
